internal/pkg/cli/job_list.go

```diff
@@ -0,0 +1,134 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package cli
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/aws/copilot-cli/internal/pkg/config"
+	"github.com/aws/copilot-cli/internal/pkg/term/prompt"
+	"github.com/aws/copilot-cli/internal/pkg/term/selector"
+	"github.com/aws/copilot-cli/internal/pkg/workspace"
+	"github.com/spf13/cobra"
+)
+
+const (
+	jobListAppNamePrompt = "Which application's jobs would you like to list?"
+)
+
+type listJobOpts struct {
+	listWkldVars
+
+	// Dependencies
+	store store
+	ws    wsJobDirReader
+	w     io.Writer
+	sel   appSelector
+}
+
+func newListJobOpts(vars listWkldVars) (*listJobOpts, error) {
+	store, err := config.NewStore()
+	if err != nil {
+		return nil, err
+	}
+	ws, err := workspace.New()
+	if err != nil {
+		return nil, err
+	}
+	return &listJobOpts{
+		listWkldVars: vars,
+
+		store: store,
+		ws:    ws,
+		w:     os.Stdout,
+		sel:   selector.NewSelect(prompt.New(), store),
+	}, nil
+}
+
+func (o *listJobOpts) Ask() error {
+	if o.appName != "" {
+		return nil
+	}
+
+	name, err := o.sel.Application(jobListAppNamePrompt, wkldListAppNameHelp)
+	if err != nil {
+		return fmt.Errorf("select application name: %w", err)
+	}
+	o.appName = name
+	return nil
+}
+
+// Execute lists the jobs in the workspace or application.
+func (o *listJobOpts) Execute() error {
+	// Ensure the application actually exists before we try to list its services.
+	if _, err := o.store.GetApplication(o.appName); err != nil {
+		return fmt.Errorf("get application: %w", err)
+	}
+
+	jobs, err := o.store.ListJobs(o.appName)
+	if err != nil {
+		return err
+	}
+
+	if o.shouldShowLocalWorkloads {
+		localNames, err := o.ws.JobNames()
+		if err != nil {
+			return fmt.Errorf("get local job names: %w", err)
+		}
+		jobs = filterSvcsByName(jobs, localNames)
```

Comment:
Then it can also be called into selector/selector.go to more easily find the intersection between workspace workloads and store workloads for the workspace selector, right?

Reply:
I think what would be super super cool is if we start creating a separate package for each command's Execute functionality.

For example, `job  ls` and `svc ls` are essentially the same code within Execute and thanks to @bvtujo 's effort pieces like `humanOutput` are re-used across commands. But we still have `o.store.GetApplication` that's common, as well as the filtering local workloads piece.

I think long-term to truly be DRY we need to create packages like:
`listing`
`packaging`
`deploy`

and the "Execute" methods just call `listing.Jobs()`, `listing.Services()`.

What do y'all think?